src/Pages/Dashboard/Dashboard.js

```diff
@@ -283,47 +257,53 @@ class DashboardLayout extends Component {
         }
 
         switch (elementType) {
-            case ('map'):
-                el = <CustomMap />;
-                break;
-            case ('linechart'):
-                el = (
-                    <LineChart
-                        data={lineChartDataset}
-                        title="Gráfico de Linhas"
-                    />
-                );
-                break;
-            case ('barchart'):
-                el = (
-                    <BarChart
-                        data={barChartDataset}
-                        title="Gráfico de Barras"
-                    />
-                );
-                break;
-            case ('piechart'):
-                el = (
-                    <PieChart
-                        data={pieChartDataset}
-                        title="Gráfico de Pizza"
-                    />
-                );
-                break;
-            case ('radarchart'):
-                el = (
-                    <RadarChart
-                        data={lineChartDataset}
-                        title="Gráfico de Radar"
-                    />
-                );
-                break;
-            case ('table'):
-                el = <EnhancedSimpleTable data={[]} header={simpleHeader} promiseData={Users.getPlainUsers} />;
-                break;
-            default:
-                el = <br />;
-                break;
+        case ('map'):
+            el = <CustomMap />;
+            break;
+        case ('linechart'):
+            el = (
+                <LineChart
+                    data={lineChartDataset}
+                    title="Gráfico de Linhas"
+                />
+            );
+            break;
+        case ('barchart'):
+            el = (
+                <BarChart
+                    data={barChartDataset}
+                    title="Gráfico de Barras"
+                />
+            );
+            break;
+        case ('piechart'):
+            el = (
+                <PieChart
+                    data={pieChartDataset}
+                    title="Gráfico de Pizza"
+                />
+            );
+            break;
+        case ('radarchart'):
+            el = (
+                <RadarChart
+                    data={lineChartDataset}
+                    title="Gráfico de Radar"
+                />
+            );
+            break;
+        case ('table'):
+            el = (
+                <EnhancedSimpleTable
+                    data={[]}
+                    header={simpleHeader}
+                    promiseData={Users.getPlainUsers}
+                />
+            );
+            break;
+        default:
+            el = <br />;
+            break;
```

Comment:
It would be interesting to replace this conditional switch.
One option is to create an object in a separate module, for example 'widgets', where the object's attributes had the same name as the 'Case' clause and the value represented JSX. Thus, it would be enough to use reflection on the object passing the ElementType (which is currently being passed to the Switch expression) to retrieve the correct JSX. Switch Default could be used if ElementType was not found via reflection in the widgets object.
Thinking that in future the widgets collection could be expanded, the widgets object could implement a [protocol Iteration](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols) (and if necessary in the future, be converted to a more complex [Generator Function](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Statements/function*)) to return the available ElementTypes so we could use a 'for .. .of 'to iterate through ElementTypes and populate the MenuItem list within FormControl.

Reply:
Nice suggestion. I will create the new Component, Widget to achieve it

Reply:
Done